iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/stack.inl

```diff
@@ -45,8 +46,10 @@ inline bool stack<T, Capacity>::push(Targs&&... args) noexcept
         return false;
     }
 
-    // low level memory management with access to the topmost element on the untyped buffer;
-    // type safety is ensured by the template parameter
+    // AXIVION Next Construct CertC++-MEM54, CertC++-EXP36, AutosarC++19_03-A18.5.10,
+    // FaultDetection-IndirectAssignmentOverflow : every entry of m_data is aligned to alignof(T)
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : low level memory management with access to the topmost element on
```

Comment:
AutosarC++19_03-A5.2.4: same justification as above. reinterpret_cast is safe as the alignment matches the alignment of T and there is sufficient space to construct a T (by design).

Reply:
Removed `reinterpret_cast`.